vic/drivers/classic/src/read_initial_model_state.c

```diff
@@ -688,4 +692,90 @@ read_initial_model_state(FILE            *init_state,
             }
         }
     }
+
+    /******Check that soil moisture does not exceed maximum allowed************/
```

Comment:
I simply moved this section from the old initialize_model_state() function.  This way, the new vic_populate_model_state() function on the classic side is much more analogous to the same function on the shared_image side.